### PR TITLE
Fix rinkeby ERC20 gas price

### PR DESCRIPTION
**Pull request by jiivan**
fixes: #1754

### CODE REVIEW

**Review by mfranciszkiewicz — COMMENTED** [review]
IMO we should use the existing `gas_price_override`

File: core/payment-driver/erc20/src/erc20/ethereum.rs [hunk 1]
```diff
@@ -187,10 +187,15 @@ pub async fn prepare_raw_transaction(
...
         .map_err(GenericError::new)?;
 
-    //get gas price from network in not provided
-    let gas_price = match gas_price_override {
-        Some(gas_price_new) => gas_price_new,
-        None => client.eth().gas_price().await.map_err(GenericError::new)?,
+    let gas_price = match network {
+        Network::Rinkeby => U256::from(1),
+        _ => {
+            //get gas price from network in not provided
```
Comment (maaktweluit) [inline core/payment-driver/erc20/src/erc20/ethereum.rs:193]:
```suggestion
            // get gas price from network if not provided
```

**Review by maaktweluit — COMMENTED** [review]
LGTM!

small wording fix proposed

**Comment by maaktweluit** [comment]
> IMO we should use the existing `gas_price_override`

i agree with @mfranciszkiewicz on this, maybe we can try to implement it properly and use this PR as a fallback / plan B?

**Review by scx1332 — DISMISSED** [review]

File: core/payment-driver/erc20/src/erc20/wallet.rs [hunk 2]
```diff
@@ -141,27 +141,36 @@ pub async fn make_transfer(
...
-    let gas_price = match gas_price {
-        Some(gas_price) => big_dec_gwei_to_u256(gas_price)?,
-        None => convert_float_gas_to_u256(get_polygon_starting_price()),
-    };
-    let max_gas_price = match max_gas_price {
-        Some(max_gas_price) => big_dec_gwei_to_u256(max_gas_price)?,
-        None => convert_float_gas_to_u256(get_polygon_maximum_price()),
+
+    let (gas_price, max_gas_price) = match network {
+        Network::Polygon => (
```
Comment (jiivan) [inline core/payment-driver/erc20/src/erc20/wallet.rs:146]: Should this apply to Mumbai also?

Comment (maaktweluit) [reply core/payment-driver/erc20/src/erc20/wallet.rs:146]: if possible i would double check with @scx1332 

i *think* it should apply to mumbai too

**Review by maaktweluit — APPROVED** [review]
LGTM!

Please check mumbai before merging :)

File: core/payment-driver/erc20/src/erc20/wallet.rs (hunk 2 above)
Comment (scx1332) [reply core/payment-driver/erc20/src/erc20/wallet.rs:146]: On Mumbai you can duplicate Rinkeby behavior

**Review by mfranciszkiewicz — APPROVED** [review]

**Pull request merged**